metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -716,6 +722,47 @@ class MetalsLanguageServer(
     onChange(paths).asJava
   }
 
+  private def insertPackage(path: AbsolutePath): Unit = {
+    def createPackageName(path: AbsolutePath): Option[String] = {
+      if (path.toFile.length() == 0) {
+        val sourceDirectory = buildTargets.inverseSourceDirectory(path)
+        val relativeDirectory = sourceDirectory
+          .map(path.toRelative)
+          .flatMap(relativePath => Option(relativePath.toFile.getParentFile))
+          .map(_.getPath)
+        val packageName = relativeDirectory.map(_.replace("/", "."))
+
+        packageName.map("package " + _)
+      } else {
+        None
+      }
+    }
+
+    def createPackageWorkspaceEdit(packageName: String): WorkspaceEdit = {
```

Comment:
packageName -> packageEditInput ? package name is actually `package` + the name

Reply:
Maybe I can move adding `package` keyword to `createWorkspaceEdit` function?

```Scala
val textEdit = new TextEdit(
    new Range(new Position(0, 0), new Position(0, 0)),
    s"package $packageName\n\n"
)
```
It would make more sense imo.